diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0823f08..abb16d4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,6 +5,8 @@
 cmake_minimum_required (VERSION 3.0.2) # 3.0.2 required for return() statement used in AddDllCopyCommands.cmake
 project(armnn)
 
+SET(CMAKE_INSTALL_PREFIX "$ENV{ARMNN_TARGET_DIR}")
+
 set(additional_cmake_files)
 list(APPEND additional_cmake_files
     cmake/Utils.cmake
@@ -73,7 +75,7 @@ if(BUILD_CAFFE_PARSER)
         set_source_files_properties(${CAFFE_GENERATED_SOURCES}/caffe/proto/caffe.pb.cc PROPERTIES COMPILE_FLAGS "-Wno-conversion -Wno-sign-conversion")
     endif()
 
-    add_library_ex(armnnCaffeParser SHARED ${armnn_caffe_parser_sources})
+    add_library_ex(armnnCaffeParser "$ENV{CK_ARMNN_LINKING_MODE}" ${armnn_caffe_parser_sources})
     set_target_properties(armnnCaffeParser PROPERTIES COMPILE_FLAGS "${CAFFE_PARSER_ADDITIONAL_COMPILE_FLAGS}")
 
     target_include_directories(armnnCaffeParser PRIVATE src/armnnUtils)
@@ -98,7 +100,7 @@ if(BUILD_ONNX_PARSER)
         set_source_files_properties(${ONNX_GENERATED_SOURCES}/onnx/onnx.pb.cc PROPERTIES COMPILE_FLAGS "-Wno-conversion -Wno-sign-conversion")
     endif()
 
-    add_library_ex(armnnOnnxParser SHARED ${armnn_onnx_parser_sources})
+    add_library_ex(armnnOnnxParser "$ENV{CK_ARMNN_LINKING_MODE}" ${armnn_onnx_parser_sources})
 
     target_include_directories(armnnOnnxParser PRIVATE src/armnnUtils)
 
@@ -121,7 +123,7 @@ if(BUILD_TF_PARSER)
         set_source_files_properties(${TF_PROTOBUFS} PROPERTIES COMPILE_FLAGS "-Wno-conversion -Wno-sign-conversion")
     endif()
 
-    add_library_ex(armnnTfParser SHARED ${armnn_tf_parser_sources})
+    add_library_ex(armnnTfParser "$ENV{CK_ARMNN_LINKING_MODE}" ${armnn_tf_parser_sources})
 
     target_include_directories(armnnTfParser PRIVATE src/armnnUtils)
 
@@ -139,7 +141,7 @@ if(BUILD_TF_LITE_PARSER)
         src/armnnTfLiteParser/TfLiteParser.cpp
         )
 
-    add_library_ex(armnnTfLiteParser SHARED ${armnn_tf_lite_parser_sources})
+    add_library_ex(armnnTfLiteParser "$ENV{CK_ARMNN_LINKING_MODE}" ${armnn_tf_lite_parser_sources})
 
     target_include_directories(armnnTfLiteParser PRIVATE src/armnn)
     target_include_directories(armnnTfLiteParser PRIVATE src/armnnUtils)
@@ -169,7 +171,7 @@ if(BUILD_ARMNN_SERIALIZER)
         src/armnnDeserializer/Deserializer.cpp
         )
 
-    add_library_ex(armnnSerializer SHARED ${armnn_serializer_sources})
+    add_library_ex(armnnSerializer "$ENV{CK_ARMNN_LINKING_MODE}" ${armnn_serializer_sources})
 
     target_include_directories(armnnSerializer PRIVATE src/armnn)
     target_include_directories(armnnSerializer PRIVATE src/armnnUtils)
@@ -409,7 +411,7 @@ foreach(lib ${armnnLibraries})
     list(APPEND armnn_sources $<TARGET_OBJECTS:${lib}>)
 endforeach()
 
-add_library_ex(armnn SHARED ${armnn_sources})
+add_library_ex(armnn "$ENV{CK_ARMNN_LINKING_MODE}" ${armnn_sources})
 
 target_include_directories(armnn PRIVATE src/armnn)
 target_include_directories(armnn PRIVATE src/armnnUtils)
